arangod/Aql/Functions.cpp

```diff
@@ -2847,6 +2847,57 @@ AqlValue Functions::DateDaysInMonth(arangodb::aql::Query* query,
   return AqlValue(AqlValueHintUInt(static_cast<uint64_t>(unsigned(lastMonthDay.day()))));
 }
 
+/// @brief function DATE_TRUNC
+AqlValue Functions::DateTrunc(arangodb::aql::Query* query,
+                            transaction::Methods* trx,
+                            VPackFunctionParameters const& parameters) {
+  static char const* AFN = "DATE_TRUNC";
+  using namespace std::chrono;
+  using namespace date;
+
+  tp_sys_clock_ms tp;
+
+  if (!ParameterToTimePoint(query, trx, parameters, tp, AFN, 0)) {
+    return AqlValue(AqlValueHintNull());
+  }
+
+  AqlValue durationType = ExtractFunctionParameterValue(parameters, 1);
+
+  if (!durationType.isString()) { // unit type must be string
+    RegisterInvalidArgumentWarning(query, AFN);
+    return AqlValue(AqlValueHintNull());
+  }
+
+  std::string duration = durationType.slice().copyString();
+  std::transform(duration.begin(), duration.end(), duration.begin(), ::tolower);
+
+  year_month_day ymd{floor<days>(tp)};
+  auto day_time = make_time(tp - sys_days(ymd));
+  milliseconds ms{0};
+
+  if (duration == "y" || duration == "year" || duration == "years") {
+    ymd = year{ymd.year()}/1/1;
```

Comment:
why divide by 1 two times?

Reply:
https://github.com/HowardHinnant/date/wiki/Examples-and-Recipes#obtaining-a-time_point-from-ymd-hms-components demonstrates how the library works with dates and operator overloading.